docs: release notes v0.11.12 (#5148)
scalameta/metals: website/blog/2023-04-14-aluminium.md

Reviewer: You might want to run prettify to break down longer lines. It's nicer to review then

---
author: Katarzyna Marek
title: Metals v0.11.12 - Aluminium
authorURL: https://github.com/kasiaMarek
authorImageURL: https://github.com/kasiaMarek.png
---

> We're happy to announce the release of Metals v0.11.12, which 

Reviewer: Could you ad some teaser or short summary?

<table>
<tbody>
  <tr>
    <td>Commits since last release</td>
    <td align="center">86</td>
  </tr>
  <tr>
    <td>Merged PRs</td>
    <td align="center">72</td>
  </tr>
    <tr>
    <td>Contributors</td>
    <td align="center">17</td>
  </tr>
  <tr>
    <td>Closed issues</td>
    <td align="center"></td>
  </tr>
  <tr>
    <td>New features</td>
    <td align="center"></td>
  </tr>
</tbody>
</table>

For full details: [https://github.com/scalameta/metals/milestone/_num_?closed=1]
(https://github.com/scalameta/metals/milestone/_num_?closed=1)

Metals is a language server for Scala that works with VS Code, Vim, Emacs and
Sublime Text. Metals is developed at the
[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
with the help from [Lunatech](https://lunatech.com) along with contributors from
the community.

## TL;DR

Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
give Metals a try!

> - [Introduce Error Reports](#introduce-error-reports)

Reviewer: ```suggestion
- [Error Reports](#introduce-error-reports)
```

- [Code action for converting dependencies from sbt to mill style](#code-action-for-converting-dependencies-from-sbt-to-mill-style)

> - [Extracting shared mtags without scalameta dependency](#extracting-shared-mtags-without-scalameta-dependency)

Reviewer: This is internal, we try to keep anything not user facing out of release notes

- [Improvements to semantic highlighting](#improvements-to-semantic-highlighting)

> - [Discovering ignored tests in test explorer](#discovering-ignored-tests-in-test-explorer)

Reviewer: I usually have a `Miscellaneous` for smaller fixes and features that don't require much explanation, I would keep it there and add any other smaller fixes

## Introduce Error Reports

> In this release we have added error reports to metals, which are ment for users to attach to github issue for additional information. Upon an error/incorrect state metals will locally create an error report in `.metals/.reports` in the workspace. For all reports we anonymize paths.

Reviewer: ```suggestion
From this release, metals will automatically generate error reports, which can later be used by users to attach to github issue for additional information. Upon any error or incorrect state metals will create an error report in `.metals/.reports` directory within the workspace. All the reports have their paths anonymised.
```

> Currently we create two types of reports:

Reviewer: ```suggestion
Currently, we create two types of reports:
```

 - incognito (under `.metals/.reports/metals`):

>   - those error reports should not contain any sensitive/private information

Reviewer: ```suggestion
  - those error reports contain only metals related stacktraces and should not contain any sensitive/private information
```

(https://github.com/scalameta/metals/milestone/56?closed=1)

Metals is a language server for Scala that works with VS Code, Vim, Emacs and
Sublime Text. Metals is developed at the
[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
with the help from [Lunatech](https://lunatech.com) along with contributors from
the community.

## TL;DR

Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
give Metals a try!

- [Error Reports](#introduce-error-reports)
- [Code action for converting dependencies from sbt to mill style](#code-action-for-converting-dependencies-from-sbt-to-mill-style)
- [Improvements to semantic highlighting](#improvements-to-semantic-highlighting)
- [Miscellaneous](#miscellaneous)

## Error Reports

From this release, metals will automatically generate error reports, which can later be used by users to attach to github issue for additional information. Upon any error or incorrect state metals will create an error report in `.metals/.reports` directory within the workspace. All the reports have their paths anonymised.

Currently, we create two types of reports:

> - incognito (under `.metals/.reports/metals`):
> - those error reports contain only metals related stacktraces and should not contain any sensitive/private information

Reviewer: ```suggestion
- _incognito_ under `.metals/.reports/metals`: those error reports contain only metals related stacktraces and should not contain any sensitive/private information
```

> - unsanitized (under `.metals/.reports/metals-full`):
> - those error reports may contain come sensitive/private information (e.g. code snippets)

Reviewer: ```suggestion
- _unsanitized_  under `.metals/.reports/metals-full`: those error reports may contain come sensitive/private information (e.g. code snippets)
```
otherwise it looks like list with 4 points.

Author: Yeah, sorry, the other two were supposed to be intended.

To ease attaching reports to github issues the `zip-reports` command zips all the **incognito** reports into `.metals/.reports/report.zip`, unsanitized reports have to be browsed and chosen by hand. Not to clutter too much, up to 30 last reports of each kind are kept at a time.

> When running into an problem in VSCode a user can use the `Metals: Open new github issue` command. This will create a template for issue with all basic info such as Metals version, used build server etc.. Next, you can browse through reports and drag and drop chosen ones to your GitHub issue. Invoking `Metals: Zip reports` command will create a zip of all the incognito reports, that you may attach too.

Reviewer: ```suggestion
When running into an problem in VSCode users can use the `Metals: Open new github issue` command. This will create a template for issue with all basic info such as Metals version, used build server etc.. Next, you can browse through reports and drag and drop chosen ones to your GitHub issue. Invoking `Metals: Zip reports` command will create a zip of all the incognito reports, that you may attach too.
```

![Reports](https://i.imgur.com/wBwFjpZ.gif)
![Zip reports](https://i.imgur.com/YN3U3N9.gif)

## Code action for converting dependencies from sbt to mill style

> For scala-cli `using (dep | lib | plugin)` directives is allows for converting dependencies from sbt style to mill style. A cool contribution by [majk-p](https://github.com/majk-p).

Reviewer: ```suggestion
New coe action for scala-cli `using (dep | lib | plugin)` directives, which allows to convert dependencies from sbt style to mill style. 

For example:
...

A great contribution by [majk-p](https://github.com/majk-p).
```

![Alt text](https://user-images.githubusercontent.com/7260594/227626189-bf6db01a-dd3e-4d0f-b20d-66ee5abc0b21.gif)

## Improvements to semantic highlighting

> A lot of improvements for semantic highlighting thanks to [jkciesluk](https://github.com/scalameta/metals/pulls/jkciesluk) and [tgodzik](https://github.com/tgodzik):

Reviewer: ```suggestion
This release brings a lot of improvements for semantic highlighting thanks to [jkciesluk](https://github.com/scalameta/metals/pulls/jkciesluk) and [tgodzik](https://github.com/tgodzik). This includes:
```

- Added declaration and definition tokens.
- Make parameters read only.
- Added semantic highlighting for using directives.
- Fixed semantic highlighting for Scala 3 worksheets.

## Miscellaneous

> - Ignored tests now show up in the test explorer (by [kpodsiad](https://github.com/kpodsiad)).
> - Improvements for package rename upon file move.
> - Fixed go to and hover for `TypeTest`.

Reviewer: ```suggestion
- bugfix: Ignored tests now show up in the test explorer (by [kpodsiad](https://github.com/kpodsiad)).
- improvement: Reworked package rename upon file move.
- bugfix: Fixed go to and hover for `TypeTest`.
```
